Dev/issue 12180 pick gdpr feature

sbreker opened the pull request:
Changes for pick to 2.4.x.  First commit is the pick from 2.5.x.  Second commit is changes required for 2.4.1 compatibility.

sbreker committed:
Add Privacy Notification GDPR,banner refs #12180

This PR includes work to add the GDPR banner. The GDPR banner can be
dismissed with the OK button - when this is performed, a session
attribute is created that will hide the banner for the session. This
works for both authenticated and unauthenticated visitors. The banner
functionality and text have a new settings panel under Admin->Settings
called "Privacy Notification" - here the GDPR banner functionality can
be deactivated, and the message can be customized. The GDPR banner will
be ON by default. Also added a new privacy message static page which is
linked from the Quick Links menu as well as from the notification banner.

Added in this commit:
- Settings page and settings sidebar menu item.
- Privacy statement static page setting.
- Quick Links menu item 'Privacy Policy'.
- Javascript that hides the GDPR message once OK button is  clicked.
- GDPR privacy message component action and template.
- GDPR controller that sets user session attribute when called from
js/privacyMessage.js.
- CSS changes to support GDPR banner message layout/formatting.

sbreker committed:
GDPR banner feature 2.4.x, refs #12180

These changes allow the GDPR banner functionality in 2.4.1.
- removed Parsedown symbols for standard html.
- since migration cannot be run, added default notification message
in case the tools:run script has not been run yet.
- removed the GDPR migration. 2.4.1 instructions will include a step
to run a tools:run script that will accomplish the same task.
- changed call to 'render_value_inline' to 'render_value'

jraddaoui commented on apps/qubit/modules/settings/actions/privacyNotificationAction.class.php:
```diff
@@ -0,0 +1,114 @@
+           ? $this->$name->getValue(array('sourceCulture' => true))
+           : $defaults[$name];
+
+         $options = array($this->i18n->__('No'), $this->i18n->__('Yes'));
```
We should normalize this some day, I prefer them this way but we have them a lot as 'yes' and 'no' (lowercase).

jraddaoui reviewed (changes requested):
Hi @sbreker,

Looking great! Good thing you made the 2.5 migration idempotent. I wonder, should we include the GDPR setup script in this PR? To void a different download in the upgrade process.

sbreker commented:
Not a bad idea including the script. I was going to post it as a gist but perhaps including it is simpler, and will not require downloading it and installing it on an AtoM server as an extra step.

Where would be a good place to put this?  It is essentially a migration, to be run using tools:run.

sbreker committed:
Add GDPR setting migration script, refs #12180

Added the 2.5.x migration as a standalone script to be run on the CLI
using tools:run.

jraddaoui reviewed (approved):
Thanks!!
